AndroidCompat/src/main/java/xyz/nulldev/androidcompat/io/sharedprefs/JavaSharedPreferences.kt

```diff
@@ -143,7 +144,10 @@ class JavaSharedPreferences(key: String) : SharedPreferences {
             itemsToAdd.forEach { (key, value) ->
                 @Suppress("UNCHECKED_CAST")
                 when (value) {
-                    is Set<*> -> preferences.encodeValue(SetSerializer(String.serializer()), key, value as Set<String>)
+                    is Set<*> -> preferences.putString(
+                        key,
+                        json.encodeToString((value as Set<String>).toList())
```

Comment:
KotlinX supports encoding/decoding Set<>
```suggestion
                        json.encodeToString(value as Set<String>)
```